lib/passwd.ml

```diff
@@ -58,16 +58,28 @@ let from_passwd_t_opt = function
   | None -> None
   | Some pw -> Some (from_passwd_t !@pw)
 
-let to_passwd_t pw =
-  let pw_t : passwd_t structure = make passwd_t in
-  setf pw_t pw_name (pw.name |> string_to_char_array |> CArray.start);
-  setf pw_t pw_passwd (pw.passwd |> string_to_char_array |> CArray.start);
-  setf pw_t pw_uid (Unsigned.UInt32.of_int pw.uid);
-  setf pw_t pw_gid (Unsigned.UInt32.of_int pw.gid);
-  setf pw_t pw_gecos (pw.gecos |> string_to_char_array |> CArray.start);
-  setf pw_t pw_dir (pw.dir |> string_to_char_array |> CArray.start);
-  setf pw_t pw_shell (pw.shell |> string_to_char_array |> CArray.start);
-  pw_t
+module Mem : sig
+  type mem
+  val to_mem : t -> mem
+  val passwd_addr_of_mem : mem -> (passwd_t, [`Struct]) Ctypes.structured Ctypes.ptr
+end = struct
+  type mem = passwd_t structure * char carray * char carray
+
+  let to_mem pw =
+    let name = string_to_char_array pw.name in
+    let passwd = string_to_char_array pw.passwd in
+    let pw_t : passwd_t structure = make passwd_t in
+    setf pw_t pw_name (CArray.start name);
+    setf pw_t pw_passwd (CArray.start passwd);
+    setf pw_t pw_uid (Unsigned.UInt32.of_int pw.uid);
+    setf pw_t pw_gid (Unsigned.UInt32.of_int pw.gid);
+    setf pw_t pw_gecos (pw.gecos |> string_to_char_array |> CArray.start);
```

Comment:
Do you need the lifetime wrapping for these fields too?

Reply:
oops, yes

Reply:
Oh yes, on passwd we need to track also `gecos`, `dir`, and `shell`.